src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -1,53 +1,62 @@
-import { Component, Input, OnChanges } from '@angular/core';
-
-import _ from 'lodash';
+import { Component, Input, ElementRef, OnInit } from '@angular/core';
+import { CssVars } from '../../enum/css-style-variable.enum';
+import { CssHelper } from '../../classes/css-helper';
 
 @Component({
   selector: 'cd-usage-bar',
   templateUrl: './usage-bar.component.html',
   styleUrls: ['./usage-bar.component.scss']
 })
-export class UsageBarComponent implements OnChanges {
-  @Input()
-  total: number;
-  @Input()
-  used: any;
-  @Input()
-  warningThreshold?: number;
-  @Input()
-  errorThreshold?: number;
-  @Input()
-  isBinary = true;
-  @Input()
-  decimals = 0;
-  @Input()
-  calculatePerc = true;
-  @Input()
-  title = $localize`usage`;
-  @Input()
-  customLegend?: string;
-  @Input()
-  customLegendValue?: string;
-  @Input()
-  showFreeToolTip = true;
-  @Input()
-  showMultisiteTooltip = false;
-
-  usedPercentage: number;
-  freePercentage: number;
+export class UsageBarComponent implements OnInit {
+  @Input() total: number = 0;
+  @Input() used: number = 903741891;
+  @Input() warningThreshold?: number = 0.9;
+  @Input() errorThreshold?: number = 0.95;
+  @Input() isBinary = true;
+  data: { group: string; value: number }[] = [{ group: $localize`Capacity`, value: 0 }];
+  options;
 
-  ngOnChanges() {
-    if (this.calculatePerc) {
-      this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
-      this.freePercentage = 100 - this.usedPercentage;
-    } else {
-      if (this.used) {
-        this.used = this.used.slice(0, -1);
-        this.usedPercentage = Number(this.used);
-        this.freePercentage = 100 - this.usedPercentage;
-      } else {
-        this.usedPercentage = 0;
+  constructor(private elementRef: ElementRef, private cssHelper: CssHelper) {
+    this.options = {
+      resizable: false,
+      meter: {
+        showLabels: false
+      },
+      tooltip: {
+        enabled: true
+      },
+      color: {
+        scale: {
+          Capacity: this.getCssVariableValue(CssVars.CDS_SUPPORT_INFO)
+        }
+      },
+      height: '1rem',
+      width: '12rem',
+      toolbar: {
+        enabled: false
       }
+    };
+  }
+
+  ngOnInit() {
+    let color = CssVars.CDS_SUPPORT_INFO; // Default color for 'Capacity'
+    const usedPercentage = this.calculateUsedPercentage(this.used, this.total);
+    if (this.warningThreshold >= 0 && usedPercentage >= this.warningThreshold) {
+      color = CssVars.CDS_SUPPORT_WARNING; // Warning threshold exceeded
+    }
+    if (this.errorThreshold >= 0 && usedPercentage >= this.errorThreshold) {
+      color = CssVars.CDS_SUPPORT_ERROR; // Error threshold exceeded
     }
+    this.data[0].value = this.total > 0 ? (this.used / this.total) * 100 : 0;
```

Comment:
why mul by 100 ? and how is total being set in this chart ? I see only current used value being passed as of now ?

Reply:
The meter chart has proportional attribute which is specifically for meter chart so actually we dont need to calculate this.

You can pass the `total`  to it and `used` as it is in the `value`

The meter chart will itself calculate things , we dont need to do that.

For status - danger, warn, success, there is  `status.range` property and carbon shows it in different way, it gives a status on top right corner. check this [example](https://stackblitz.com/run?file=src%2Foptions.ts)

![image](https://github.com/user-attachments/assets/79d70ede-973d-4699-a3a2-d418a78c3536)

we can follow the same instead of bootstrapping the prev behavior.

Reply:
You would need to pass the unit and `used` as it is without converting.